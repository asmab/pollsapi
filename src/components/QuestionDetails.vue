<template>
    <div class="header-content">
        <div class="row">
            <div class="col col-1"> {{index+1}} </div>
            <div class="col col-10"> 
                {{question.question}}<hr>
                <div class="question-details">
                    <span class="date"> Published at :{{dateFormatter(question.published_at)}}</span>
                    <span class="choices"> {{question.choices.length}} Choices </span>
                </div>
            </div>

            <div class="col col-1">
                <button class="vote-btn" v-b-popover.hover="'Click me to vote !'" block href="#" v-b-toggle="'num-'+index"></button>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        index: {
            required: true
        },
        question: {
            required: true
        }
    },
    methods: {
        dateFormatter(published_date){
            var convertedDate = new Date(published_date)
            return convertedDate.toLocaleDateString().toString()
        }
    }
}
</script>

<style lang="scss">

$orange-color: #f39c12 ;

hr { margin: 0px; }
.date { float:left; }
.choices { float: right; }

.question-details {
    margin-top:5px;
}

.header-content {
    text-align: left;
        color: $orange-color;
        font-size: 20px;
        font-weight: bold;
            span{
        color: white;
        font-size: 10px;
        margin:0px;
        padding:0px;
        list-style-type: none;
    }
}


.vote-btn{
    width: 30px;
    height: 30px;
    border: none;
    background-color: $orange-color;
    background: transparent; 
    background-image: url("../assets/vote.png");
    background-repeat: no-repeat;
    cursor: pointer;
}
.btn-wrapper {
    float:right;
}

</style>