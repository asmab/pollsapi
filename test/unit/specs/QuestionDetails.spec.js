import { mount } from '@vue/test-utils'
import QuestionDetails from '@/components/QuestionDetails'

describe('QuestionDetails component', () => {
    let wrapper

    const mockPropsData = { 
        index: 1,
        question : {
            "question": "Favourite programming language?",
            "published_at": "2014-11-11T08:40:51.620Z",
            "url": "/questions/1",
            "choices": [{
                "choice": "Swift",
                "url": "/questions/1/choices/1",
                "votes": 2048
            }]
        }

    }

    beforeEach(() => {
        wrapper = mount(QuestionDetails, { mockPropsData } )
    })

    it('QuestionDetails component looks as expected', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})