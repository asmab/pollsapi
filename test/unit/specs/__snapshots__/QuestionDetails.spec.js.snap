// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionDetails component QuestionDetails component looks as expected 1`] = `
<div class="header-content">
    <div class="row">
        <div class="col col-1 index"> 2 </div>
        <div class="col col-10">
            Favourite programming language?
            <hr>
            <div class="question-details"><span class="date"> Published at :11/11/2014</span> <span class="choices"> 1 Choices </span></div>
        </div>
        <div class="col col-1">
            <button block="" href="#" class="vote-btn"></button>
        </div>
    </div>
</div>
`;
